packages/astro/src/core/build/index.ts

```diff
@@ -37,24 +37,31 @@ export interface BuildOptions {
 	/**
 	 * Teardown the compiler WASM instance after build. This can improve performance when
 	 * building once, but may cause a performance hit if building multiple times in a row.
+	 *
+	 * @internal only used for testing
+	 * @default true
 	 */
 	teardownCompiler?: boolean;
 }
 
-/** `astro build` */
-export default async function build(
-	inlineConfig: AstroInlineConfig,
-	options: BuildOptions
-): Promise<void> {
+/**
+ * Builds your site for deployment. By default, this will generate static files and place them in a dist/ directory.
+ * If SSR is enabled, this will generate the necessary server files to serve your site.
+ */
+export default async function build(inlineConfig: AstroInlineConfig): Promise<void> {
 	applyPolyfill();
 	const logging = createNodeLogging(inlineConfig);
 	const { userConfig, astroConfig } = await resolveConfig(inlineConfig, 'build');
 	telemetry.record(eventCliSession('build', userConfig));
 
 	const settings = createSettings(astroConfig, fileURLToPath(astroConfig.root));
 
+	// Use this hack to prevent the build options being public API
+	// eslint-disable-next-line prefer-rest-params
+	const internalOptions = arguments[1] as BuildOptions | undefined;
```

Comment:
Note: I chose to do this trick since I didn't want a wrapper `build` function that _hides_ the option. Otherwise there's a risk the implementation diverges and the jsdoc would be in an odd place.

If anyone feels it's better to wrap it instead, I'm also open to do so. I don't have a hard preference.

Reply:
Personally I'd prefer that we wrap it. I understand the concern about implementation divergence, but we can keep both functions in this file and make sure there are comments explaining what needs to happen when updating.